src/app/storage/query/utils.ts

```diff
@@ -0,0 +1,97 @@
+import { Method } from 'axios';
+
+import { StorageResponse } from '../types/storage';
+import { fetchStorage } from '../utils';
+
+import { SingleQueryResponse, StoredQueriesResponse, StoredQueryObject } from './types';
+
+export const needsQueryId = (method: Method) => ['DELETE', 'PUT'].includes(method);
+
+export const initialiseQueryStorage = async (etag: string): Promise<StorageResponse | void> => {
+  if (etag) {
+    return fetchStorage({
+      data: JSON.stringify([
+        {
+          op: 'add',
+          path: '/queries',
+          value: {},
+        },
+      ]),
+      etag,
+      method: 'PATCH',
+    });
+  }
+
+  throw {
+    message: 'no etag was present while attempting to initialise query storage',
+    status: 400,
+  };
+};
+
+export const fetchQueries =
```

Comment:
gets specifically the queries part of the storage, and initialises it if missing.